### ISSUE TITLE
MD011: false positive for parenthesis plus footnote

### ISSUE THREAD

**Issue by ndarville** [markdownlint/markdownlint#132]
``` markdown
(...)[^footnote]
```

Is regarded as a reverse link, because markdownlint considers footnote syntax (part Markdown Extra) a link.

**Comment by singledoggy** [comment]
Also happens when cross ref bibliography when writting academic paper after (). Maybe we should check the content in [] is a link
```
(somthing)[#citekey]
```
